File: relationship-manager/recommendation-service/src/main/java/com/bank/rm/recommendation/controller/RecommendationController.java
package com.bank.rm.recommendation.controller;

import com.bank.rm.common.dto.ApiResponse;
import com.bank.rm.recommendation.dto.RecommendationDtos.*;
import com.bank.rm.recommendation.service.RecommendationService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.UUID;

@RestController
@RequestMapping("/api/v1/recommendations")
public class RecommendationController {
    private final RecommendationService recommendationService;

    public RecommendationController(RecommendationService recommendationService) {
        this.recommendationService = recommendationService;
    }

    @PostMapping("/{customerId}")
    public ResponseEntity<ApiResponse<RecommendationResponse>> recommend(
            @PathVariable UUID customerId, @RequestBody RecommendationRequest request) {
        return ResponseEntity.ok(ApiResponse.ok(recommendationService.generateRecommendations(customerId, request)));
    }
}

File: relationship-manager/recommendation-service/src/main/kotlin/com/bank/rm/recommendation/controller/RecommendationController.kt
package com.bank.rm.recommendation.controller

import com.bank.rm.common.dto.ApiResponse
import com.bank.rm.recommendation.dto.*
import com.bank.rm.recommendation.service.RecommendationService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import java.util.UUID

@RestController
@RequestMapping("/api/v1/recommendations")
class RecommendationController(private val recommendationService: RecommendationService) {

    @PostMapping("/{customerId}")
    @ResponseStatus(HttpStatus.CREATED)
    fun generate(
        @PathVariable customerId: UUID,
        @Valid @RequestBody request: RecommendationRequest
    ): ApiResponse<RecommendationResponse> =
        ApiResponse(success = true, data = recommendationService.generateRecommendations(customerId, request))

    @GetMapping("/{customerId}/latest")
    fun getLatest(@PathVariable customerId: UUID): ApiResponse<RecommendationResponse?> =
        ApiResponse(success = true, data = recommendationService.getLatestRecommendation(customerId))
}
